### DISCUSSION TITLE
Have <template> x-attributes (x-for, x-if, etc) work by default inside <svg> elements

### DISCUSSION THREAD

**Discussion by ika-musuko** [alpinejs/alpine#4269] [7. Bugs (archived)]
[This polyfill](https://github.com/alpinejs/alpine/issues/637#issuecomment-654856538) from #637 allows the `<template>` x-attributes (like `x-for` and `x-if`) to work as intended inside `<svg>` elements. Particularly, I'm calling it when the `DOMContentLoaded` event is fired with `document.addEventListener`.

But you have to dig around the internet to find this snippet and patch it in yourself.

It would be great if this polyfill was included in Alpine.js by default so we don't need to patch this in manually when we want to use `<template>` x-attributes inside `<svg>`.

Another related discussion: #3944

**Comment by SimoTod** [comment]
Unfortunately it's not really an Alpine issue but it's caused by the svg specs which unfortunately are inconsistent with the standard DOM.
Many devs value the small size of the library so adding a polyfill for just a small number of users is usually avoided.
Perhaps something in the documentation would help people wasting less time (assuming they read the documentation 😂).

**Comment by ekwoka** [comment]
I vote for not including.

It can be handled in userspace easily enough, and isn't going to be used by most people.

The core needs to focus on things with high use or reasonable utility that really needs to be in core (or else it can't be done).

Also, Simo, how could you have let that code out into the world...
